jsonschema-generator/src/main/java/com/github/victools/jsonschema/generator/MethodScope.java

```diff
@@ -171,11 +177,49 @@ public boolean isGetter() {
     public <A extends Annotation> A getAnnotation(Class<A> annotationClass) {
         A annotation = super.getAnnotation(annotationClass);
         if (annotation == null) {
-            annotation = this.getRawMember().getAnnotatedReturnType().getAnnotation(annotationClass);
+            annotation = findGetterCandidates(this.getRawMember()).stream().map(m -> {
+                return Optional.ofNullable(m.getAnnotation(annotationClass))
+                        .orElse(m.getAnnotatedReturnType().getAnnotation(annotationClass));
+            }).filter(Objects::nonNull).findFirst().orElse(null);
         }
         return annotation;
     }
 
+    private Collection<Method> findGetterCandidates(Method m) {
+        List<Method> candidates = new ArrayList<>();
+        addGetterCandidates(candidates, m.getDeclaringClass(), m.getName());
+        return candidates;
+    }
+
+    private void addGetterCandidates(List<Method> candidates, Class<?> declaringClass,
+            String methodName) {
+        Method getterInClass = findGetterInClass(declaringClass, methodName);
+        if (getterInClass != null) {
+            candidates.add(getterInClass);
+        }
+        for (Class<?> i : declaringClass.getInterfaces()) {
+            Method getterInInterface = findGetterInClass(i, methodName);
```

Comment:
**question**: Would this be covering a method on an extended interface? Maybe we'd have to extend this even further?

Not sure I have considered that everywhere consistently though. Perhaps I should write some more tests to check that. 😉